bankid/asyncclient.py

```diff
@@ -1,12 +1,12 @@
-from typing import Optional, Tuple, Dict, Any
+from typing import Any, Dict, Tuple, Union
 
 import httpx
 
 from bankid.baseclient import BankIDClientBaseclass
 from bankid.exceptions import get_json_error_class
 
 
-class BankIDAsyncClient(BankIDClientBaseclass):
+class BankIDAsyncClient(BankIDClientBaseclass[httpx.AsyncClient]):
```

Comment:
Doesn't this imply that the BankIDClientBaseClass **is** a httpx.Asyncclient, which it isn't? It could have been, but I decided against it to provide a clearer separation.

Reply:
No, it doesn't imply that and I agree with the separation. Since `BankIDClientBaseclass` inherits `typing.Generic` this binds the type variable to `httpx.AsyncClient` in the context of `BankIDAsyncClient`. I would have prefered to set something like `client: httpx.BaseClient` on the base class but that kind of type isn't exposed from httpx. The only thing it really does right now (since `self.client` isn't widely used) is "preventing" us from assigning something other than an `httpx.AsyncClient` instance to `self.client`.